routing/turns.cpp

```diff
@@ -292,7 +299,8 @@ void SplitLanes(string const & lanesString, char delimiter, vector<string> & lan
 bool ParseSingleLane(string const & laneString, char delimiter, TSingleLane & lane)
 {
   lane.clear();
-  istringstream laneStream(laneString);
+  // put a delimiter after each token
+  istringstream laneStream(laneString + delimiter);
```

Comment:
BTW, do we really need to append a delimiter?
If yes, please, update TestParseSingleLane unit test with such cases.

Reply:
Yes. I pushed a new comment here to explain, but when `laneString` ended in an empty token representing `none`, the empty string would be ignored, not processed. Adding a delimiter fixes that.

There is now a corresponding case in `TestParseSingleLane` and `TestParseLanes`.